src/trio/_core/_run.py

```diff
@@ -1859,6 +1860,12 @@ def task_exited(self, task: Task, outcome: Outcome[Any]) -> None:
             assert task._parent_nursery is not None, task
             task._parent_nursery._child_finished(task, outcome)
 
+        # before or after the other stuff in this function?
+        if task in GLOBAL_PARKING_LOT_BREAKER:
+            for lot in GLOBAL_PARKING_LOT_BREAKER[task]:
+                lot.break_lot(task)
```

Comment:
IMO we should do this before the misnested-cancel-scopes check up the top; if that happens to fire we still want locks to break instead of hanging.